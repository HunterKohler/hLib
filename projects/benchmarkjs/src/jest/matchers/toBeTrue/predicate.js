export default recieved => recieved === true;