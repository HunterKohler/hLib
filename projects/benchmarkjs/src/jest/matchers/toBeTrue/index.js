import { matcherHint, printReceived, printExpected } from "jest-matcher-utils";
import predicate from "./predicate.js";

const passMessage = recieved => () =>
    matcherHint(".not.toBeTrue", "recieved", "") 
    + "\n\n"
    + "Expected value to not be true, recieved:\n"
    + `    ${printReceived(recieved)}`;

const failMessage = recieved => () =>
    matcherHint(".toBeTrue", "recieved", "")
    + "\n\n"
    + "Expected value to be true:\n"
    + `    ${printExpected(true)}\n`
    + "Recieved:\n"
    + `    ${printReceived(recieved)}`;

export default function toBeTrue(recieved) {
    const pass = predicate(recieved);
    return { pass, message: pass
        ? passMessage(recieved)
        : failMessage(recieved)
    };
}