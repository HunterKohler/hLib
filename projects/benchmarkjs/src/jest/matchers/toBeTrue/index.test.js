import toBeTrue from "./";

expect.extend({ toBeTrue });

describe(".toBeTrue", () => {
    test("pass when given true", () => {
        expect(true).toBeTrue();
    })

    test.each([
        [false], [""], [0], [1], [{}], [[]], [undefined], [null], [NaN]
    ])("fail when given %s", value => {
        expect(() => expect(value).toBeTrue()).toThrowErrorMatchingSnapshot();
    });
});

describe(".not.toBeTrue", () => {
    test("fail when given true", () => {
        expect(() => expect(true).not.toBeTrue()).toThrowErrorMatchingSnapshot();
    });

    test.each([
        [false], [""], [0], [1], [{}], [[]], [undefined], [null], [NaN]
    ])("pass when given %s", value => {
        expect(() => expect(value).not.toBeTrue());
    })
});