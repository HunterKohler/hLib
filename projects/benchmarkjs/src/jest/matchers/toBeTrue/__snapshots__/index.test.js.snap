// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toBeTrue fail when given true 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).not.toBeTrue()[22m

Expected value to not be true, recieved:
    [31mtrue[39m"
`;

exports[`.toBeTrue fail when given  1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeTrue()[22m

Expected value to be true:
    [32mtrue[39m
Recieved:
    [31m\\"\\"[39m"
`;

exports[`.toBeTrue fail when given [] 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeTrue()[22m

Expected value to be true:
    [32mtrue[39m
Recieved:
    [31m[][39m"
`;

exports[`.toBeTrue fail when given {} 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeTrue()[22m

Expected value to be true:
    [32mtrue[39m
Recieved:
    [31m{}[39m"
`;

exports[`.toBeTrue fail when given 0 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeTrue()[22m

Expected value to be true:
    [32mtrue[39m
Recieved:
    [31m0[39m"
`;

exports[`.toBeTrue fail when given 1 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeTrue()[22m

Expected value to be true:
    [32mtrue[39m
Recieved:
    [31m1[39m"
`;

exports[`.toBeTrue fail when given NaN 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeTrue()[22m

Expected value to be true:
    [32mtrue[39m
Recieved:
    [31mNaN[39m"
`;

exports[`.toBeTrue fail when given false 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeTrue()[22m

Expected value to be true:
    [32mtrue[39m
Recieved:
    [31mfalse[39m"
`;

exports[`.toBeTrue fail when given null 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeTrue()[22m

Expected value to be true:
    [32mtrue[39m
Recieved:
    [31mnull[39m"
`;

exports[`.toBeTrue fail when given undefined 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeTrue()[22m

Expected value to be true:
    [32mtrue[39m
Recieved:
    [31mundefined[39m"
`;
